modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/components/OrderableTable.tsx

```diff
@@ -0,0 +1,322 @@
+			className={classNames('orderable-table-row', {
+				dragging: isDragging,
+			})}
+			key={index}
+			ref={ref}
+		>
+			<ClayTable.Cell className="drag-handle-cell">
+				<ClayButtonWithIcon
+					aria-label={Liferay.Util.sub(
+						Liferay.Language.get('drag-x'),
+						item.label || Liferay.Language.get('item')
+					)}
+					displayType={null}
+					symbol="drag"
+				/>
+			</ClayTable.Cell>
+
+			{fields.map((field) => {
+				const itemFieldValue = String(item[field.name]);
+
+				const fuzzyMatch = fuzzy.match(
+					query,
+					itemFieldValue,
+					FUZZY_OPTIONS
+				);
+
+				return (
+					<ClayTable.Cell key={field.name}>
+						{fuzzyMatch ? (
+							<span
+								dangerouslySetInnerHTML={{
+									__html: fuzzyMatch.rendered,
+								}}
+							/>
+						) : (
+							<span>{itemFieldValue}</span>
+						)}
+					</ClayTable.Cell>
+				);
+			})}
+		</ClayTable.Row>
+	);
+};
+
+interface OrderableTableInterface {
+	fields: Array<{
+		label: string;
+		name: string;
+	}>;
+	items: Array<any>;
+	noItemsButtonLabel: string;
+	noItemsDescription: string;
+	noItemsTitle: string;
+	onCancelButtonClick: Function;
+	onCreationButtonClick: Function;
+	onOrderChange: Function;
+	onSaveButtonClick: Function;
+	title: string;
+}
+
+const OrderableTable = ({
+	fields,
+	items: initialItems,
+	noItemsButtonLabel,
+	noItemsDescription,
+	noItemsTitle,
+	onCancelButtonClick,
+	onCreationButtonClick,
+	onOrderChange,
+	onSaveButtonClick,
+	title,
+}: OrderableTableInterface) => {
+	const [items, setItems] = useState(initialItems);
+	const [query, setQuery] = useState('');
+
+	const onSearch = (query: string) => {
+		setQuery(query);
+
+		const regexp = new RegExp(query, 'i');
+
+		setItems(
+			query
+				? initialItems.filter((item) =>
+						fields.some((field) =>
+							String(item[field.name]).match(regexp)
+						)
+				  ) || []
+				: initialItems
+		);
+	};
+
+	const handleOnOrderChange = ({
+		draggedIndex,
+		targetIndex,
+	}: {
+		draggedIndex: number;
+		targetIndex: number;
+	}) => {
+		const orderedItems = items.map((item, index) => {
+			if (index === draggedIndex) {
+				return items[targetIndex];
+			}
+
+			if (index === targetIndex) {
+				return items[draggedIndex];
+			}
+
+			return item;
+		});
+
+		setItems(orderedItems);
+
+		onOrderChange({
+			orderedItems,
+		});
+	};
+
+	return (
+		<ClayLayout.Sheet className="mt-3 orderable-table-sheet">
+			<ClayLayout.SheetHeader>
+				<h2 className="sheet-title">{title}</h2>
+			</ClayLayout.SheetHeader>
+
+			<ClayLayout.SheetSection>
+				<ManagementToolbar.Container>
+					<ManagementToolbar.ItemList expand>
+						<ManagementToolbar.Item className="nav-item-expand">
+							<Search onSearch={onSearch} query={query} />
+						</ManagementToolbar.Item>
+
+						<ManagementToolbar.Item>
+							<ClayButtonWithIcon
+								aria-label={Liferay.Language.get('add')}
+								className="nav-btn nav-btn-monospaced"
+								onClick={() => onCreationButtonClick()}
```

Comment:
I think you can pass it directly through
```
onClick={onCreationButtonClick}
```

Reply:
TS complains if I do this, `Type 'Function' is not assignable to type 'MouseEventHandler<HTMLButtonElement>'.`. My thinking was that we don't really need access to `event` in `onCreationButtonClick`, it is a bit higher level abstraction in API. Maybe it makes sense as this is a reusable component? I'm not sure.